misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -0,0 +1,419 @@
+                Token bodyStartDelimiter = AbstractNodeFactory.createToken(SyntaxKind.OPEN_BRACE_TOKEN);
+                Token bodyEndDelimiter = AbstractNodeFactory.createToken(SyntaxKind.CLOSE_BRACE_TOKEN);
+                List<Node> recordFList = new ArrayList<>();
+                for (Map.Entry<String, Schema> property: properties.entrySet()) {
+                    addRecordFields(required, recordFList, property);
+                }
+                NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFList);
+                TypeDescriptorNode typeDescriptorNode = NodeFactory.createRecordTypeDescriptorNode(recordKeyWord,
+                        bodyStartDelimiter, fieldNodes, null, bodyEndDelimiter);
+                Token semicolon = AbstractNodeFactory.createToken(SyntaxKind.SEMICOLON_TOKEN);
+                TypeDefinitionNode typeDefinitionNode = NodeFactory.createTypeDefinitionNode(null,
+                        null, typeKeyWord, typeName, typeDescriptorNode, semicolon);
+                typeDefinitionNodeList.add(typeDefinitionNode);
+                Token refTypeName = AbstractNodeFactory.createIdentifierToken(StringUtils.capitalize(name));
+                return createBuiltinSimpleNameReferenceNode(null, refTypeName);
+
+            } else {
+                outStream.println("Encountered an unsupported type. Type `any` would be used for the field.");
+                Token typeName = AbstractNodeFactory.createToken(SyntaxKind.ANY_KEYWORD);
+                return createBuiltinSimpleNameReferenceNode(null, typeName);
+            }
+        } else if (schema.get$ref() != null) {
+            Token typeName = AbstractNodeFactory.createIdentifierToken(extractReferenceType(schema.get$ref()));
+            return createBuiltinSimpleNameReferenceNode(null, typeName);
+        } else {
+            //This contains a fallback to Ballerina common type `any` if the OpenApi specification type is not defined
+            // or not compatible with any of the current Ballerina types.
+            outStream.println("Encountered an unsupported type. Type `any` would be used for the field.");
+            Token typeName = AbstractNodeFactory.createToken(SyntaxKind.ANY_KEYWORD);
+            return createBuiltinSimpleNameReferenceNode(null, typeName);
+        }
+        Token typeName = AbstractNodeFactory.createToken(SyntaxKind.ANY_KEYWORD);
+        return createBuiltinSimpleNameReferenceNode(null, typeName);
+    }
+
+    /**
+     * Parse and get the {@link OpenAPI} for the given json Schema String contract.
+     *
+     * @param schemaString     json Schema as a string
+     * @return {@link OpenAPI}  OpenAPI model
+     * @throws ConverterException in case of invalid schema
+     * @throws IOException in case of Json parse error
+     */
+    public static OpenAPI parseJSONSchema(String schemaString) throws ConverterException, IOException {
+        final String prefix = "{\n" +
```

Comment:
In case of a CRLF input file, here the end of line characters will be mixed in our `openAPIFileContent` string, i.e. `\n` in the prefix and suffix, and `\r\n` in the middle. But as we are just passing this string into the `OpenAPIV3Parser()` this will not have an effect on the output if the parser can handle an input with mixed line endings. I added a test in 6cfbf95 with CRLF formatted json file and it seems to be fine with that.